### PR TITLE
Adding start/stop functionality from the UI

### PR DESCRIPTION
**Pull request by JamesWP**
## Description

adds the start stop functionality requested/mentioned  in issue #1586

adds the -nsc --no-search-control option to disable this feature (on by default)

## Motivation and Context

allows user to start stop the service without accessing the server

addresses issue: #1586

## How Has This Been Tested?
click the stop button, service stops cleanly
click the start button, starts again
clicking either button when already started / stopped does nothing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

added new endpoint in app.py to accept status update requests
added methods to control the search thread and to support restarting
once stopped
added option to sidebar for user controls

### CODE REVIEW

File: pogom/search.py [hunk 1]
```diff
@@ -192,3 +196,14 @@ def search_loop(args):
         log.info('Crashed, waiting {:d} seconds before restarting search.'.format(args.scan_delay))
         time.sleep(args.scan_delay)
         search_loop(args)
+
+def search_loop_stop():
+    global stopping
+    """
+        stops the searching method after curent operation has completed
```
Comment (tegob) [inline pogom/search.py]: typo: current

File: templates/map.html [hunk 2]
```diff
@@ -125,6 +125,13 @@
 					</label>
 				</div>
 			</div>
+			<div class="form-control switch-container">
+				<h3>Search Control&nbsp;</h3>
+        <p id="search-status"></p>
+			  <button id="stop-search">Stop Searching</button>
+        <br/>
```
Comment (tegob) [inline templates/map.html]: Please follow the existing coding style and use `<br>` instead of `<br/>`.

File: templates/map.html [hunk 3]
```diff
@@ -125,6 +125,13 @@
 					</label>
 				</div>
 			</div>
+			<div class="form-control switch-container">
+				<h3>Search Control&nbsp;</h3>
+        <p id="search-status"></p>
```
Comment (tegob) [inline templates/map.html]: Indentation

**Comment by tegob** [comment]
LGTM with nits addressed

File: templates/map.html (hunk 2 above)
Comment (JamesWP) [inline templates/map.html]: bugger, didn't notice the tabs not spaces :(

File: templates/map.html (hunk 3 above)
Comment (JamesWP) [inline templates/map.html]: bugger, didn't notice the tabs not spaces :(

File: templates/map.html [hunk 4]
```diff
@@ -125,6 +125,13 @@
 					</label>
 				</div>
 			</div>
+      <div class="form-control switch-container">
```
Comment (tegob) [inline templates/map.html]: Hmm, indentation is still weird. Looks like you’re using spaces instead of tabs

**Comment by JamesWP** [comment]
Just looking into how this can be configured to be disabled

**Comment by mariuskvarner** [comment]
Tested, works great!

**Comment by JamesWP** [comment]
Also addresses #1828

**Comment by joshslark** [comment]
lol, I was reading this right after I posted it

**Comment by JamesWP** [comment]
If you could test and report back would be great

**Comment by joshslark** [comment]
It works great in the web app, but the log is cluttered with multiple "Status" messages that don't seem to serve a purpose.

**Comment by espro** [comment]
Bugged in Firefox for me, Stop/Start buttons show up at the same time, and the Start button is clipped, cannot scroll further.

Image of clipped button

**Comment by mariuskvarner** [comment]
Can somebody try running it together with -ns? Breaks it for me.

**Comment by JamesWP** [comment]
@espro the buttons are always shown they are not supposed to disappear
i will take a look at the scrolling sidebar

**Comment by JamesWP** [comment]
@espro  I believe i have fixed the style issue, not an issue with my code but something that became visible

**Comment by JamesWP** [comment]
it still prevents the server stopping with ^C when under -ns, anyone have a clue why?

**Comment by joshslark** [comment]
The way search_thread is defined seemed to cause it. In master, line 22 is "search_thread = Thread()". Defining it this way allowed ctrl-c to stop the server while resolving the scope issue. I also passed args as a parameter to avoid defining it globally.

**Comment by JamesWP** [comment]
I have changed it to include the search_thead in the top of the file and it still wont close :/

**Comment by JamesWP** [comment]
fixed the issue with the server not closing as it was supposed to do

**Comment by joshslark** [comment]
I'm guessing the reason it still wouldn't close was because search_thread was still defined as a global variable under start_locator_thread function. It works in its current state in my repository, if you want to see what was different when I commented.

**Comment by JamesWP** [comment]
Yeah send it over


Sent from my Samsung device

-------- Original message --------

**Comment by joshslark** [comment]
My repo is at [https://github.com/joshslark/PokemonGo-Map](url) , but other than that I don't know of any way to send it over. I can't create a pull request with your repo as a base fork.

**Comment by JamesWP** [comment]
I'll take a look when I get time




Sent from my Samsung device

-------- Original message --------

**Comment by JamesWP** [comment]
either way, the problem is fixed now. Ready for merge

**Comment by mackhankins** [comment]
Just what I was looking for.

**Comment by JamesWP** [comment]
Go ask the merge gods to bump here and they or I can get this to a point where we can merge this

**Comment by mackhankins** [comment]
@JamesWP I love the feature, but from your fork it's still clipping for me.  Also, would this be better served as a toggle instead of buttons?

Works aces though.

**Comment by JamesWP** [comment]
Yeah wouldn't be too hard to adapt that way either.

If you want you can make the change and submit a new PR to the main project or my fork, either way

**Comment by mackhankins** [comment]
@JamesWP We would be outside my depth at that point.  However, the clipping can be fixed by adding a class to _nav.scss like 

```css
		.search-control {
			margin-top: 10px;
		}
````

**Comment by mackhankins** [comment]
@JamesWP I updated it to the latest develop branch.  Do you want me to send a new pull request?https://github.com/mackhankins/PokemonGo-Map/tree/develop

screen shot 2016-07-28 at 7 59 07 pm

**Comment by chuyskywalker** [comment]
I am keeping an eye on this as it's a useful feature, but it does need to be rebased to merge cleanly still, and have a few fixes it seems.

**Comment by chuyskywalker** [comment]
I also agree that this would work better as a toggle; it's a lot of UI otherwise for the same functionality.

**Comment by mackhankins** [comment]
@chuyskywalker Okay, thanks for the response.  My limited skills with python couldn't get me to a toggle.  Maybe @JamesWP can knock it out.

**Comment by JamesWP** [comment]
If anyone is interested Im working on a new version of this PR here:
https://github.com/JamesWP/PokemonGo-Map
I'll post here when I have completed the PR

**Comment by JamesWP** [comment]
Ok i have finally found the time to create a new up to date PR for this: please go show support etc here if you want it merged in

**Pull request closed without merging**